### PR TITLE
docs: add docs page for how to parameterize your tests

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: docs/src/test-parameterize-js.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+---
+id: test-parameterize
+title: "Parameterize tests"
+---
+
+You can either parameterize tests on a project level or for single tests on file level.
```
Comment (pavelfeldman) [inline docs/src/test-parameterize-js.md:6]:
```suggestion
You can either parameterize tests on a test level or on a project level.
```

File: docs/src/test-parameterize-js.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+<!-- TOC -->
+
+## Tests
```
Comment (pavelfeldman) [inline docs/src/test-parameterize-js.md:10]:
```suggestion
## Parametrized Tests
```

File: docs/src/test-parameterize-js.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+const people = ['Alice', 'Bob'];
+for (const name in people) {
+  test(`testing with ${name}`, async () => {
+    // ...
+  });
+  // You can also do it with test.describe() or with multiple tests as long the test name is unique.
+}
+```
+
+## Projects
```
Comment (pavelfeldman) [inline docs/src/test-parameterize-js.md:34]:
```suggestion
## Parametrized Projects
```

File: docs/src/test-parameterize-js.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+module.exports = config;
+```
+
+```js js-flavor=ts
+// playwright.config.ts
+import { PlaywrightTestConfig } from '@playwright/test';
+import { TestOptions } from './my-test';
+
+const config: PlaywrightTestConfig<TestOptions> = {
```
Comment (pavelfeldman) [inline docs/src/test-parameterize-js.md:113]: Let's fix our story here, it looks too complex. I think we should do something like:

```js
declare namespace TestOptions {
  person: string;
}
```

```js
use: { person: 'Alice' }
```

File: docs/src/test-parameterize-js.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+```js js-flavor=ts
+// my-test.ts
+import { test as base } from '@playwright/test';
+
+export type TestOptions = {
+  person: string;
+};
+
+export const test = base.extend<TestOptions>({
+  // Default value for the version.
```
Comment (JoelEinbinder) [inline docs/src/test-parameterize-js.md:58]:
```suggestion
  // Default value for the person.
```

**Comment by pavelfeldman** [comment]
You can now save following as `testParameters.d.ts` to get the project parametrization in place.
```js
export {};

declare global {
  namespace PlaywrightTest {
    export interface ExtraUseOptions {
      parameter: string;
    }  
  }
}
```

**Pull request merged**